### DISCUSSION TITLE
What's the proper way to typed the `items` prop for dynamic collections?

### DISCUSSION THREAD

**Discussion by brandonpittman** [adobe/react-spectrum#3863] [Q&A]
I found this example in the docs, but it's just punting the types and using `any[]`. 

https://codesandbox.io/s/heuristic-margulis-uz3d4d?file=/src/App.tsx

**Comment by snowystinger** [comment]
Are you running into an issue if you just do something like
`let items: {id: string, name: string}[] = []`?
We don't know what your items will be, it's up to you to type them.

**Reply by brandonpittman** [reply]
@snowystinger 

I may be losing something when I wrap my listbox inside my custom select. 

<img width="792" alt="CleanShot 2022-12-21 at 09 53 59@2x" src="https://user-images.githubusercontent.com/967145/208795192-ba30f3d1-a160-46fd-a8f6-48cef84f9560.png">

The `Item` component always types my `item` as `object`.

**Reply by snowystinger** [reply]
Yeah, something seems off, because I see it just fine in the codesandbox with the change I suggested
<img width="476" alt="Screenshot 2022-12-21 at 10 50 06 AM" src="https://user-images.githubusercontent.com/698229/208960158-afd4c9dc-30ad-4f9a-93ec-d212de35036b.png">

Can you share more of your code? Maybe we'll spot the difference there. You can also look at our source code, it's all typed as well.
